sycl/include/sycl/handler.hpp

```diff
@@ -2870,6 +2893,25 @@ class __SYCL_EXPORT handler {
   void memcpyFromDeviceGlobal(void *Dest, const void *DeviceGlobalPtr,
                               bool IsDeviceImageScoped, size_t NumBytes,
                               size_t Offset);
+
+  template <typename T, int Dims, access::mode AccessMode,
+            access::target AccessTarget,
+            access::placeholder IsPlaceholder = access::placeholder::false_t,
+            typename PropertyListT = property_list>
+  void checkIfPlaceholderIsBoundToHandler(
+      accessor<T, Dims, AccessMode, AccessTarget, IsPlaceholder, PropertyListT>
+          Acc) {
+    detail::AccessorBaseHost *AccBase = (detail::AccessorBaseHost *)&Acc;
+    detail::AccessorImplPtr AccImpl = detail::getSyclObjImpl(*AccBase);
```

Comment:
Can we not call `detail::getSyclObjImpl` on `Acc` directly?

Reply:
It seems like no, as
`detail::AccessorImplPtr AccImpl = detail::getSyclObjImpl(Acc);`
leads to compilation error:

> /localdisk2/nkornev/llvm/build/bin/../include/sycl/detail/common.hpp:300:42: note: candidate template ignored: substitution failure [with Obj = accessor<int, 1, (mode)1024, (target)2014, (placeholder)0, accessor_property_list<>>]: 'impl' is a protected member of 'sycl::accessor<int, 1, sycl::access::mode::read, sycl::access::target::global_buffer, sycl::access::placeholder::false_t, sycl::ext::oneapi::accessor_property_list<>>'
> template <class Obj> decltype(Obj::impl) getSyclObjImpl(const Obj &SyclObject) {